bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    harness::{read_to_bytes, Mode, TlsBenchHarness},
+    CA_CERT_PATH, SERVER_CERT_CHAIN_PATH, SERVER_KEY_PATH,
+};
+use log::debug;
+use s2n_tls::{
+    callbacks::VerifyHostNameCallback,
+    config::{Builder, Config},
+    connection::Connection,
+    enums::Blinding,
+    security::DEFAULT_TLS13,
+};
+use std::{
+    cell::UnsafeCell,
+    collections::VecDeque,
+    ffi::c_void,
+    io::{Read, Write},
+    os::raw::c_int,
+    pin::Pin,
+    task::Poll::Ready,
+};
+
+pub struct SignalToNoise {
+    // connections need to share *mut buffers for custom IO so we use UnsafeCell
+    // to make the UnsafeCell not move around in memory, we wrap it in Pin<Box<T>>
```

Comment:
I would phrase this as "IO buffers are mutated by both client and server and therefore need to be of type UnsafeCell. Additionally the buffer is wrapped in Pin<Box<T>> to ensure  long-term pointers can access it"

Reply:
If both client and server need to mutate IO buffers, a `RefCell` would also work, but I think the main justification for the `UnsafeCell` is that we share `*mut` pointers; hopefully my edited comment is clear enough.